Foundation/URLSession/Message.swift

```diff
@@ -49,24 +49,24 @@ extension _NativeProtocol._ParsedResponseHeader {
     /// that ending.
     /// - Returns: Returning nil indicates failure. Otherwise returns a new
     ///     `ParsedResponseHeader` with the given line added.
-    func byAppending(headerLine data: Data) -> _NativeProtocol._ParsedResponseHeader? {
+    func byAppending(headerLine data: Data, headerCompleted: (String) -> Bool) -> _NativeProtocol._ParsedResponseHeader? {
```

Comment:
Please document the parameters in the comments above the function.

Reply:
I think the new parameter would be better named `onHeaderCompleted` to indicate that it's a callback and not a simple boolean.

Reply:
It's not a boolean value though - it's a closure?

Reply:
Yeah I meant the closure which does the check

Reply:
The parameter name refers to the whole parameter though, so it should make it clear that the parameter is a closure, not a boolean. `headerCompleted` sounds like it's a boolean that is being passed in ("is the header already completed?")